posix/devices/hostfs/hostfs.c

```diff
@@ -781,11 +788,68 @@ static int _hostfs_fcntl(oe_fd_t* desc, int cmd, uint64_t arg)
 {
     int ret = -1;
     file_t* file = _cast_file(desc);
+    void* argout = NULL;
+    uint64_t argsize = 0;
 
     if (!file)
         OE_RAISE_ERRNO(OE_EINVAL);
 
-    if (oe_posix_fcntl_ocall(&ret, file->host_fd, cmd, arg) != OE_OK)
+    switch (cmd)
+    {
+        case OE_F_GETFD:
+        case OE_F_SETFD:
+        case OE_F_GETFL:
+        case OE_F_SETFL:
+            break;
+
+        case OE_F_GETLK:
+        case OE_F_OFD_GETLK:
+            argsize = sizeof(struct oe_flock);
+            argout = (void*)arg;
+            break;
+
+        case OE_F_SETLKW:
+        case OE_F_SETLK:
+        {
+            void* srcp = (void*)arg;
+            argsize = sizeof(struct oe_flock);
+            argout = (void*)arg;
+            memcpy(argout, srcp, argsize);
```

Comment:
Same comments as the previous file that had the switch statement for fcntl.

Reply:
Same resolution. WI think since fcntl has a finite set of possible values, those values should be reflected in the switch.  Ioctl is somewhat different, since there are literally an infinite number of those, potentially